src/server/test/db/cikVaryTests.js

```diff
@@ -0,0 +1,150 @@
+
+	// For 10 -> 1
+
+	// Split at 2020-01-01 00:00:00
+	await ConversionSegment.splitLater(
+		unit10Id, unit1Id, null, 2, 0, null,
+		'-infinity', 'infinity', '2020-01-01 00:00:00', conn
+	);
+
+	// Split at 2020-06-01 00:00:00 (on the segment starting at 2020-01-01)
+	await ConversionSegment.splitLater(
+		unit10Id, unit1Id, null, 3, 0, null,
+		'2020-01-01 00:00:00', 'infinity', '2020-06-01 00:00:00', conn
+	);
+
+	// Split at 2021-01-01 00:00:00 (on the segment starting at 2020-06-01)
+	//await ConversionSegment.splitLater(
+	//	unit10Id, unit1Id, null, 4, 0, null,
+	//	'2020-06-01 00:00:00', 'infinity', '2021-01-01 00:00:00', conn
+	//);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 6, 0, null,
+		'-infinity', 'infinity', '2020-01-01 00:00:00', conn
+	);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 7, 0, null,
+		'2020-01-01 00:00:00', 'infinity', '2020-06-01 00:00:00', conn
+	);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 8, 0, null,
+		'2020-06-01 00:00:00', 'infinity', '2021-01-01 00:00:00', conn
+	);
+}
+
+// Insert test conversions using the Conversion class
+async function insertTestConversions(conn) {
+	const unit10Id = (await Unit.getByName('Unit 10', conn)).id;
+	const unit1Id = (await Unit.getByName('Unit 1', conn)).id;
+	const unit2Id = (await Unit.getByName('Unit 2', conn)).id;
+	const conversionPreInsert1 = new Conversion(unit10Id, unit1Id, false, 'note');
+	const conversionPreInsert2 = new Conversion(unit1Id, unit2Id, false, 'note');
+	await Promise.all([
+		conversionPreInsert1.insert(null, 1, 0, "", conn),
+		conversionPreInsert2.insert(null, 1, 0, "notes", conn)
+	]);
+}
+
+mocha.describe('CIK Vary Chaining', () => {
+
+	mocha.describe('Insert test unit', function () {
+		mocha.it('should insert units without error', async function () {
+			const conn = testDB.getConnection();
+			await insertTestUnits(conn);
+			const units = await conn.any('SELECT * FROM units');
+			expect(units).to.have.lengthOf(3);
+		});
+	});
+	mocha.describe('Insert test conversions', function () {
+		mocha.it('should insert conversions without error', async function () {
+			const conn = testDB.getConnection();
+			await insertTestUnits(conn);
+			await insertTestConversions(conn);
+			const conversions = await conn.any('SELECT * FROM conversions');
+			expect(conversions).to.have.lengthOf(2);
+		});
+	});
+
+	mocha.describe('Insert test segments', () => {
+		mocha.it('should insert segments without error', async () => {
+			const conn = testDB.getConnection();
+			await insertTestUnits(conn);
+			await insertTestConversions(conn);
+			await insertTestSegments(conn);
+			const segments = await conn.any('SELECT * FROM conversion_segments');
+			expect(segments).to.have.lengthOf(7);
+		});
+	});
+
+	mocha.describe('Update cik_vary and views', function () {
+		mocha.it('should update cik_vary with the correct amount of segments', async function () {
```

Comment:
I can see the value of having separate tests while these were developed. However, in many OED tests, it first checks the length and then tests each item. This means these two tests could be combined. It would also mean that the current second test would also test length before items which is a stronger test.